src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ExpressionLambdaRewriter.cs

```diff
@@ -957,7 +957,7 @@ private BoundExpression VisitObjectCreationExpressionInternal(BoundObjectCreatio
 
             if ((object)node.Constructor == null ||
                 (node.Arguments.Length == 0 && !node.Type.IsStructType()) ||
-                node.Constructor.IsDefaultValueTypeConstructor())
+                node.Constructor.IsDefaultValueTypeConstructor(requireZeroInit: false))
```

Comment:
Looks like expression trees scenarios are handled already.
Could you clarify the follow-up work item tracked in test plan? "Expression tree: synthesized parameterless constructor is not called in Expression e = () => new S();"

Reply:
This call should have used `requireZeroInit: true`. Fixed and checked off work item in test plan.